### ISSUE TITLE
Dynamic Background Image Error

### ISSUE THREAD

**Issue by sambbaron** [dudaster/ele-custom-skin#33]
I have the latest versions of Elementor (2.5.16) and Ele Custom Skin installed (1.1.0). With the recent update for dynamic background images, I am getting the following error:
>  Illegal string offset 'background_image' in ...\wp-content\plugins\elementor\includes\base\controls-stack.php on line 1174

I am testing with a very simple skin that only has a section with the background image set as the dynamic featured image and a heading widget with the post title. The post title renders, but the image does not. Thank you.

**Comment by dudaster** [comment]
EPro version?

**Comment by sambbaron** [comment]
2.5.9

**Comment by danifola** [comment]
My template was working fine until I updated Elementor Skin to verion 1.1.0. I do have a background image. The error I get is:

Fatal error: Uncaught Error: Cannot iterate on string offsets by reference in /home/ikid2684/public_html/wp-content/plugins/elementor/includes/base/controls-stack.php:1153
Stack trace:
#0 /home/ikid2684/public_html/wp-content/plugins/ele-custom-skin/ele-custom-skin.php(40): Elementor\Controls_Stack->parse_dynamic_settings('', Array)
#1 /home/ikid2684/public_html/wp-includes/class-wp-hook.php(286): ECS_set_bg_element(Object(Elementor\Element_Section))
#2 /home/ikid2684/public_html/wp-includes/class-wp-hook.php(310): WP_Hook->apply_filters(NULL, Array)
#3 /home/ikid2684/public_html/wp-includes/plugin.php(465): WP_Hook->do_action(Array)
#4 /home/ikid2684/public_html/wp-content/plugins/elementor/includes/base/element-base.php(658): do_action('elementor/front...', Object(Elementor\Element_Section))
#5 /home/ikid2684/public_html/wp-content/plugins/elementor/core/base/document.php(1168): Elementor\Element_Base->print_element()
#6 /home/ikid2684/public_html/wp-content/plugins/elementor/core/base/document.php(808): Elementor\Core\ in /home/ikid2684/public_html/wp-content/plugins/elementor/includes/base/controls-stack.php on line 1153

**Comment by dudaster** [comment]
I'm having trouble reproducing this thing... what plugins themes do you have installed?

**Comment by dudaster** [comment]
please send me an email where I can login and try to fix it.

**Comment by danifola** [comment]
Hi dudaster, I sent you information so you can log on; and here's the Elementor info 
					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: Percona Server (GPL), Release 84.0, Revision 47234b3 v5.6.40-84.0
	PHP Version: 7.1.30
	PHP Max Input Vars: 1000
	PHP Max Post Size: 128M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.2.2
	Site URL: https://ikid.club
	Home URL: https://ikid.club
	WP Multisite: No
	Max Upload Size: 128 MB
	Memory limit: 768M
	Permalink Structure: /%postname%/
	Language: es
	Timezone: America/Argentina/Buenos_Aires
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elelemtor Child
	Version: 1.0
	Author: Daniele Folatelli
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.1.2
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: es_ES
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.100 Safari/537.36

== Active Plugins ==
	Antispam Bee
		Version: 2.9.1
		Author: pluginkollektiv

	Anywhere Elementor
		Version: 1.1
		Author: WebTechStreet

	BackWPup
		Version: 3.6.9
		Author: Inpsyde GmbH

	Code Snippets
		Version: 2.13.3
		Author: Shea Bunge

	Custom Icons for Elementor
		Version: 0.2.3
		Author: Michael Bourne

	DynamicConditions
		Version: 1.2.1
		Author: RTO GmbH

	Dynamic Visibility for Elementor
		Version: 3.0.1
		Author: Dynamic.ooo

	Ele Custom Skin
		Version: 1.1.0
		Author: Liviu Duda

	Elementor
		Version: 2.5.16
		Author: Elementor.com

	Elementor Pro
		Version: 2.5.9
		Author: Elementor.com

	Forminator
		Version: 1.7.0.1
		Author: WPMU DEV

	GamiPress
		Version: 1.7.4.2
		Author: GamiPress

	Google Analytics for WordPress by MonsterInsights
		Version: 7.7.1
		Author: MonsterInsights

	Nav Menu Roles
		Version: 1.9.3
		Author: Kathy Darling

	Paid Memberships Pro
		Version: 2.0.7
		Author: Stranger Studios

	Pods - Custom Content Types and Fields
		Version: 2.7.12
		Author: Pods Framework Team

	Query Monitor
		Version: 3.3.6
		Author: John Blackbourn

	SG Optimizer
		Version: 5.1.3
		Author: SiteGround

	Show Current Template
		Version: 0.3.0
		Author: JOTAKI Taisuke

	Smush
		Version: 3.2.1
		Author: WPMU DEV

	Tabulate
		Version: 2.10.3
		Author: Sam Wilson

	Theme My Login
		Version: 7.0.14
		Author: Theme My Login

	User Role Editor
		Version: 4.51.1
		Author: Vladimir Garagulya

	Wordfence Security
		Version: 7.3.4
		Author: Wordfence

	Wordpress File Upload
		Version: 4.11.2
		Author: Nickolas Bossinas

	Yoast SEO
		Version: 11.5
		Author: Team Yoast


== Log ==
	: 
JS: showing 17 of 17JS: 2019-05-17 15:10:49 [error X 2][https://ikid.club/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.5.15:2:165400] t is undefined 
JS: 2019-05-17 15:31:30 [error X 7][https://ikid.club/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.5.15:2:213040] e.ui.form is undefined 
JS: 2019-05-23 19:38:43 [error X 3][https://c0.wp.com/c/5.2.1/wp-includes/js/jquery/jquery.js line 2 > eval:12:5] jQuery(...).datepicker is not a function 
JS: 2019-05-26 16:05:07 [error X 1][https://ikid.club/wp-content/plugins/dynamic-visibility-for-elementor/assets/js/dce-editor.js?ver=2.0.4:44:26] Can\'t find variable: elementorFrontend 
JS: 2019-05-26 19:35:56 [error X 16][https://ikid.club/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.5.15:2:47913] undefined is not an object (evaluating \'t.element\') 
JS: 2019-05-27 17:48:39 [error X 4][https://ikid.club/wp-content/plugins/elementor/assets/js/editor-modules.min.js?ver=2.5.15:2:9579] An invalid argument supplied as multiple control value [array (
  'category' => 'InvalidElementData',
  'tag' => 'Element `undefined` got <https://ikid.club/wp-content/plugins/essential-addons-for-elementor-lite//assets/front-end/img/accordion.png> as `eael_accordion_bg` value. Expected array or object.',
)]
JS: 2019-05-27 18:44:31 [error X 1][https://ikid.club/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5:22:19954] View (cid: \"view65280\") has already been destroyed and cannot be used. 
JS: 2019-05-27 18:44:31 [error X 1][https://ikid.club/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5:22:19954] View (cid: \"view65358\") has already been destroyed and cannot be used. 
JS: 2019-05-27 18:44:31 [error X 1][https://ikid.club/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5:22:19954] View (cid: \"view65389\") has already been destroyed and cannot be used. 
JS: 2019-05-27 19:05:26 [error X 1][https://ikid.club/wp-content/plugins/dynamic-visibility-for-elementor/assets/js/dce-editor.js?ver=2.0.4:44:9] elementorFrontend is not defined 
JS: 2019-05-30 14:51:31 [error X 2][https://ikid.club/wp-content/plugins/dynamic-visibility-for-elementor/assets/js/dce-editor.js?ver=2.0.4:117:9] elementorFrontend.config.elements.data[cid] is undefined 
JS: 2019-06-05 20:00:47 [error X 3][https://ikid.club/wp-content/plugins/elementor/assets/js/editor-modules.min.js?ver=2.5.16:2:10531] n.attributes[t.name].each is not a function 
JS: 2019-06-05 20:00:48 [error X 1][https://ikid.club/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5:22:19954] View (cid: \"view49211\") has already been destroyed and cannot be used. 
JS: 2019-06-20 21:27:45 [error X 3][https://ikid.club/wp-content/plugins/dynamic-visibility-for-elementor/assets/js/dce-editor-visibility.js?ver=3.0.1:87:27] Can\'t find variable: dce_enable_save_button 
JS: 2019-06-21 13:12:53 [error X 2][https://ikid.club/wp-content/plugins/dynamic-visibility-for-elementor/assets/js/dce-editor-visibility.js?ver=3.0.1:87:5] dce_enable_save_button is not defined 
JS: 2019-06-22 13:42:02 [error X 3][https://ikid.club/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.5.16:2:75476] setting getter-only property \"type\" 
JS: 2019-06-25 17:46:34 [error X 1][https://ikid.club/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5:22:19954] View (cid: \"view410\") has already been destroyed and cannot be used. 

Log: showing 1 of 12019-05-28 12:33:03 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.5.15',
  'to' => '2.5.16',
)]

PHP: showing 6 of 6PHP: 2019-05-31 18:39:59 [notice X 4][/home/ikid2684/public_html/wp-content/plugins/elementor-pro/modules/theme-builder/widgets/post-content.php::114] Trying to get property of non-object [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2019-06-03 13:07:23 [notice X 24][/home/ikid2684/public_html/wp-content/plugins/elementor-pro/modules/dynamic-tags/pods/tags/pods-base.php::31] Undefined index: evidencia [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2019-06-05 17:05:56 [error X 2][/home/ikid2684/public_html/wp-content/plugins/elementor/core/documents-manager.php::470] Uncaught Error: Call to a member function is_built_with_elementor() on boolean in /home/ikid2684/public_html/wp-content/plugins/elementor/core/documents-manager.php:470
Stack trace:
#0 [internal function]: Elementor\Core\Documents_Manager->ajax_save(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#1 /home/ikid2684/public_html/wp-content/plugins/elementor/core/common/modules/ajax/module.php(171): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 /home/ikid2684/public_html/wp-includes/class-wp-hook.php(286): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#3 /home/ikid2684/public_html/wp-includes/class-wp-hook.php(310): WP_Hook->apply_filters('', Array)
#4 /home/ikid2684/public_html/wp-includes/plugin.php(465): WP_Hook->do_action(Array)
#5 /home/ikid2684/public_html/wp-admin/admin-ajax.php(173): do_action('wp_ajax_element...')
#6 {main}
  thrown [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2019-06-18 03:54:38 [notice X 38][/home/ikid2684/public_html/wp-content/plugins/elementor-pro/modules/dynamic-tags/pods/tags/pods-text.php::37] Array to string conversion [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2019-06-25 14:59:22 [error X 5][/home/ikid2684/public_html/wp-content/plugins/elementor/includes/base/controls-stack.php::1153] Uncaught Error: Cannot iterate on string offsets by reference in /home/ikid2684/public_html/wp-content/plugins/elementor/includes/base/controls-stack.php:1153
Stack trace:
#0 /home/ikid2684/public_html/wp-content/plugins/ele-custom-skin/ele-custom-skin.php(40): Elementor\Controls_Stack->parse_dynamic_settings('', Array)
#1 /home/ikid2684/public_html/wp-includes/class-wp-hook.php(286): ECS_set_bg_element(Object(Elementor\Element_Section))
#2 /home/ikid2684/public_html/wp-includes/class-wp-hook.php(310): WP_Hook->apply_filters(NULL, Array)
#3 /home/ikid2684/public_html/wp-includes/plugin.php(465): WP_Hook->do_action(Array)
#4 /home/ikid2684/public_html/wp-content/plugins/elementor/includes/base/element-base.php(658): do_action('elementor/front...', Object(Elementor\Element_Section))
#5 /home/ikid2684/public_html/wp-content/plugins/elementor/core/base/document.php(1168): Elementor\Element_Base->print_element()
#6 /home/ikid2684/public_html/wp-content/plugins/elementor/core/base/document.php(808): Elementor\Core\ [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2019-06-25 16:17:45 [error X 14][/home/ikid2684/public_html/wp-content/plugins/elementor/includes/base/controls-stack.php::1153] Uncaught Error: Cannot iterate on string offsets by reference in /home/ikid2684/public_html/wp-content/plugins/elementor/includes/base/controls-stack.php:1153
Stack trace:
#0 /home/ikid2684/public_html/wp-content/plugins/ele-custom-skin/ele-custom-skin.php(40): Elementor\Controls_Stack->parse_dynamic_settings('A', Array)
#1 /home/ikid2684/public_html/wp-includes/class-wp-hook.php(286): ECS_set_bg_element(Object(Elementor\Element_Section))
#2 /home/ikid2684/public_html/wp-includes/class-wp-hook.php(310): WP_Hook->apply_filters(NULL, Array)
#3 /home/ikid2684/public_html/wp-includes/plugin.php(465): WP_Hook->do_action(Array)
#4 /home/ikid2684/public_html/wp-content/plugins/elementor/includes/base/element-base.php(658): do_action('elementor/front...', Object(Elementor\Element_Section))
#5 /home/ikid2684/public_html/wp-content/plugins/elementor/core/base/document.php(1168): Elementor\Element_Base->print_element()
#6 /home/ikid2684/public_html/wp-content/plugins/elementor/core/base/document.php(808): Elementor\Core [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

**Comment by dudaster** [comment]
Thanks @danifola !

I solved the problem and the fix would be available tomorrow in version 1.1.1.

@danifola you already have the fix on your website.

**Comment by danifola** [comment]
Great! Thank you very much!

**Comment by TheCrystalRose** [comment]
I'm experiencing a fatal error since the last update as well... It's happening on every page I have a Loop template. What do I do?

![Homepage Fatal Error](https://user-images.githubusercontent.com/39744470/60219025-20abc380-983f-11e9-8146-f5e3161815c0.PNG)

**Comment by TheCrystalRose** [comment]
Nevermind. I did some investigating and realized that Essential Addon's plugin had the Parallex option enabled, which caused the interference between Ele Skin and Elementor. Is there a fix for this, by the way?

**Comment by dudaster** [comment]
The new version should fix this too. It would be available in the next 24 hours.

**Comment by TheCrystalRose** [comment]
Okay, excellent! Looking forward to the new version!

**Comment by dudaster** [comment]
Please update and let me know!

**Comment by sambbaron** [comment]
Thanks for the quick turnaround! So the update does add the background image, but there is another issue. The additional background image CSS properties (position, repeat, size) are not being rendered. They are working in edit/preview. Here are some screenshots:

![Ele Custom Skin - Preview](https://user-images.githubusercontent.com/10456742/60282040-e7189e00-98ba-11e9-9fe6-13e902957529.PNG)
![Ele Custom Skin - Render](https://user-images.githubusercontent.com/10456742/60282047-ea138e80-98ba-11e9-9251-5cf453b5b430.PNG)

**Comment by dudaster** [comment]
All I am changing is the url propriety the rest should be inherited. You should be able to scroll down and see them applied. I’m not overwriting them.

**Comment by sambbaron** [comment]
I checked all the CSS properties on the `<section>` - the other background properties are not there.

**Comment by dudaster** [comment]
Is there a way I can take a look myself?

**Comment by sambbaron** [comment]
I've only got this project running locally right now. Were you able to reproduce?

**Comment by dudaster** [comment]
this is my playground for this plugin: https://clean.ehot.ro/

**Comment by sambbaron** [comment]
Okay, I see the other background properties on your site. Not sure what's going on with my site. It works in edit. I thought maybe it was because I have styles set to inline, but changing to external file didn't matter.

**Comment by dudaster** [comment]
but If you don’t use dynamic image is it working?

**Comment by sambbaron** [comment]
If I don't use a dynamic image, it does work

**Comment by dudaster** [comment]
Please try to find a way to make it public so I can get access to it. Meanwhile are you using other elementor plugins? What theme?

**Comment by sambbaron** [comment]
I'm using the Hello Elementor theme with a child theme. Here are my plugins:
```
Advanced Custom Fields | Version 5.8.1 by Elliot Condon
Custom Post Type UI | Version 1.6.2 by WebDevStudios
Defender | Version 2.1.2 by WPMU DEV
Duplicator | Version 1.3.14 by Snap Creek
Ele Conditions | Version 1.0.5 by Liviu Duda
Ele Custom Skin | Version 1.1.2 by Liviu Duda
Elementor | Version 2.5.16 by Elementor.com
Elementor Blocks for Gutenberg | Version 1.0.1 by Elementor.com
Elementor Pro | Version 2.5.9 by Elementor.com
Enable Media Replace | Version 3.3.1 by ShortPixel
Media Library Folders for WordPress | Version 5.0.2 by Max Foundry
Regenerate Thumbnails Advanced | Version 2.1.0 by ShortPixel
Search & Replace | Version 3.2.1 by Inpsyde GmbH
ShortPixel Image Optimizer | Version 4.14.1 by ShortPixel
UpdraftPlus - Backup/Restore | Version 1.16.15 by UpdraftPlus.Com, DavidAnderson
Windows Compatibility Fix | Version 1.0.2 by Dave Jesch
WP-Optimize | Version 2.3.4 by David Anderson, Ruhani Rabin, Team Updraft
WP Fastest Cache | Version 0.8.9.5 by Emre Vona
WP Mail SMTP | Version 1.4.2 by WPForms
```

**Comment by dudaster** [comment]
could be the cache plugins?

**Comment by dudaster** [comment]
please send me an email with your skype id so we can fix this fast. I taking to long this way.

**Comment by sambbaron** [comment]
I copied the project and removed the custom theme and all non-Elementor plugins. This did not fix the problem - when using a dynamic image (featured image), the background css properties (position, repeat, size) are still not rendering on the page.

You can access this project here:
https://test.neworiginscoaching.com
username: abcd
password: 1234

I only have a single site Elementor Pro license right now while I'm evaluating Elementor, so you will have to switch to your license. Happy to Skype if you think that would be helpful; let me know and I'll PM you. Thank you.

**Comment by sambbaron** [comment]
I copied the project and removed the custom theme and all non-Elementor plugins. This did not fix the problem - when using a dynamic image (featured image), the background css properties (position, repeat, size) are still not rendering on the page. I deployed the site publicly, and I'll email you with access details.

**Comment by dudaster** [comment]
Something fishy is happening. I installed latest beta version and it seems to fixed the problem. 

The issue surfaces when you change the template the last css is deleted and it only creates it when you refresh the page and it doesn't do it properly... I don't know the reason why.

It worked when I made the changes in the loop template and then refresh it's css... first I received no file error and the second refresh was working corectly.

I don't know how and why this behaviour, this was not happening on neither of my servers. 

Try reinstall elementor ... and make sure it is clean and untempered with.
